scripts/validation-utils.js

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env node
+
+const shell = require('shelljs');
+shell.set('-e');
+shell.set('+v');
+
+const NODE_VERSION = '12.4.0';
+const LERNA_VERSION = '3.13.1';
+
+module.exports = {
+
+  checkVSCodeVersion: () => {
+    const nextVersion = process.env['SALESFORCEDX_VSCODE_VERSION'];
+    if (!nextVersion || !nextVersion.match(/^(\d+)\.(\d+)\.(\d+)$/)) {
+      console.log(`You must set environment variable 'SALESFORCEDX_VSCODE_VERSION'.`);
+      console.log(
+        `To set: 'export SALESFORCEDX_VSCODE_VERSION=xx.yy.zz'. Where xx.yy.zz is the release number.`
+      );
+      process.exit(-1);
+    }
+  },
+
+  checkCircleCiToken: () => {
+    if (!process.env['CIRCLECI_TOKEN']) {
+      console.log(`You must set environment variable 'CIRCLECI_TOKEN'.`);
+      console.log(
+        `To set: 'export CIRCLECI_TOKEN=token'. This token allows us to pull the artifacts that will be released.`
+      );
+      process.exit(-1);
+    }
+  },
+
+  checkCircleCiBuild: () => {
+    if (!process.env['CIRCLECI_BUILD']) {
+      console.log(`You must set environment variable 'CIRCLECI_BUILD'.`);
+      console.log(
+        `To set: 'export CIRCLECI_BUILD=build_num'. Where build_num contains the artifacts that will be released.`
+      );
+      process.exit(-1);
+    }
+  },
+
+  checkEnvironmentVariables: () => {
+    console.log('\nVerifying environment variables have been set.');
+    module.exports.checkVSCodeVersion();
+    module.exports.checkCircleCiToken();
+    module.exports.checkCircleCiBuild();
+  },
+
+  checkNodeVerion: () => {
+    console.log('\nVerifying node version ' + NODE_VERSION + ' is installed.');
+    const [version, major, minor, patch] = process.version.match(/^v(\d+)\.?(\d+)\.?(\*|\d+)$/);
+    if (parseInt(major) != NODE_VERSION.split('.')[0] || parseInt(minor) < NODE_VERSION.split('.')[1]) {
+      console.log('Please update from node version ' + process.version + ' to ' + NODE_VERSION);
+      process.exit(-1);
+    }
+  },
+
+  checkLernaInstall: () => {
+    console.log(`\nVerifying lerna is installed for node version ${NODE_VERSION}.`);
+    if (!shell.which('lerna') || !shell.which('lerna').includes(NODE_VERSION)) {
+      console.log('Lerna not found - Installing lerna version ' + LERNA_VERSION);
+      shell.exec('npm install -g lerna@' + LERNA_VERSION);
+    }
+  },
+
+  checkVSCEInstall: () => {
+    console.log(`\nVerifying vsce is installed for node version ${NODE_VERSION}.`);
+    if (!shell.which('vsce') || !shell.which('vsce').includes(NODE_VERSION)) {
```

Comment:
The issue Jason had run into last week around the vsce installation was due to switching node versions. He had to reinstall vsce for node version 12.4.0.

Reply:
To expand - this checks that the installation of the current vsce matches the version of node that we require.